frontend/src/App.svelte

```diff
@@ -138,11 +142,23 @@
           meta: {
             persistent: true,
           },
-        });      
+        });
       }
     }
   }
-  
+
+  cacheDir.subscribe((cacheDirectory) => {
+    GetCacheDirDiskSpaceLeft().then((spaceLeftBytes) => {
+      if (spaceLeftBytes < 10e9) {
+        const spaceLeftGbReadable = (spaceLeftBytes * 1e-9).toFixed(1);
+        $error = `The drive your cache directory is on (${cacheDirectory}) is very low on disk space (Only ~${spaceLeftGbReadable} GB left). Please free up some space or move the cache directory to another drive in the Mod Manager Settings.`;
+      }
+    }).catch((err) => {
+      $error = `failed to check cache directory disk space left: ${err}`;
+    });
+  });
+
+
```

Comment:
I got the idea to do this from the cacheLocationPicker where it uses subscribe without returning the unsubscribe (https://github.com/satisfactorymodding/SatisfactoryModManager/blob/32e3b76ff7c61c45ff5714975ad3dd466bad2076/frontend/src/lib/components/modals/settings/cacheLocationPicker.ts#L6)

Can you explain what `$` is making reactive in this situation? I thought it wouldn't know what to watch to re-call GetCacheDirDiskSpaceLeft()

Reply:
`$: ` is the magical keyword for svelte to look for `$store`s, finding `$cacheDir`, so it will retrigger the whole section pointed to by `$: ` whenever that changes. `$: ` isn't something that happens at runtime, Svelte gets compiled to efficient js based on its defined keywords

Reply:
Is there a reason you changed it to 10e100 in the suggestion?

Reply:
Ah, nope, that was for testing the dialog and I didn't notice it when I copied it here